src/main/java/org/spongepowered/common/mixin/core/network/MixinPacketBuffer.java

```diff
@@ -0,0 +1,372 @@
+import org.spongepowered.asm.mixin.Implements;
+import org.spongepowered.asm.mixin.Interface;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.common.network.ByteBufUtils;
+import org.spongepowered.common.service.persistence.NbtTranslator;
+
+import java.io.IOException;
+import java.nio.ByteOrder;
+import java.util.UUID;
+
+@Mixin(PacketBuffer.class)
+@Implements(@Interface(iface = ChannelBuf.class, prefix = "cbuf$"))
+public abstract class MixinPacketBuffer extends ByteBuf {
+
+    @Shadow private ByteBuf buf;
+
+    @Shadow
+    protected abstract NBTTagCompound readNBTTagCompoundFromBuffer() throws IOException;
+
+    @Shadow
+    protected abstract void writeNBTTagCompoundToBuffer(NBTTagCompound compound);
+
+    private ChannelBuf oppositeOrder;
+
+    public int cbuf$getCapacity() {
+        return this.buf.capacity();
+    }
+
+    public int cbuf$available() {
+        return this.buf.writerIndex() - this.buf.readerIndex();
+    }
+
+    public ChannelBuf cbuf$order(ByteOrder order) {
+        checkNotNull(order, "order");
+        if (this.buf.order().equals(order)) {
+            return (ChannelBuf) this;
+        }
+        if (this.oppositeOrder == null) {
+            this.oppositeOrder = (ChannelBuf) new PacketBuffer(this.buf.order(order));
+        }
+        return this.oppositeOrder;
+    }
+
+    public ByteOrder cbuf$getByteOrder() {
+        return this.buf.order();
+    }
+
+    public int cbuf$readerIndex() {
+        return this.buf.readerIndex();
+    }
+
+    public ChannelBuf cbuf$setReadIndex(int index) {
+        this.buf.readerIndex(index);
+        return (ChannelBuf) this;
+    }
+
+    public int cbuf$writerIndex() {
+        return this.buf.writerIndex();
+    }
+
+    public ChannelBuf cbuf$setWriteIndex(int index) {
+        this.buf.writerIndex(index);
+        return (ChannelBuf) this;
+    }
+
+    public ChannelBuf cbuf$setIndex(int readIndex, int writeIndex) {
+        this.buf.setIndex(readIndex, writeIndex);
+        return (ChannelBuf) this;
+    }
+
+    public ChannelBuf cbuf$clear() {
+        this.buf.clear();
+        return (ChannelBuf) this;
+    }
+
+    public ChannelBuf cbuf$markRead() {
+        this.buf.markReaderIndex();
+        return (ChannelBuf) this;
+    }
+
+    public ChannelBuf cbuf$markWrite() {
+        this.buf.markWriterIndex();
+        return (ChannelBuf) this;
+    }
+
+    public ChannelBuf cbuf$resetRead() {
+        this.buf.resetReaderIndex();
+        return (ChannelBuf) this;
+    }
+
+    public ChannelBuf cbuf$resetWrite() {
+        this.buf.resetWriterIndex();
+        return (ChannelBuf) this;
+    }
+
+    public ChannelBuf cbuf$slice() {
+        return (ChannelBuf) new PacketBuffer(this.buf.slice());
+    }
+
+    public ChannelBuf cbuf$slice(int index, int length) {
+        return (ChannelBuf) new PacketBuffer(this.buf.slice(index, length));
+    }
+
+    public byte[] cbuf$array() {
```

Comment:
All methods use the underlying buf field so there shouldn't be any self-referencing methods anywhere